### ISSUE TITLE
Portable app conversion

### ISSUE THREAD

**Issue by Liger0** [visualboyadvance-m/visualboyadvance-m#679]
### Expected behavior
It would be useful to have the program in a portable version with his own .ini settings file, so you can have various versions of the software as needed and each one configured differently, also being able to use the program on the fly with usb drives

### Actual behavior
settings .ini file is stored in appdata/local

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.): Windows 10

**Comment by ZachBacon** [comment]
It is portable. So what you do is remove the settings file from appdata, and then you move it to the exe folder and it will use that. That's been a thing since forever

**Comment by Liger0** [comment]
Why not doing it by default or making it as a selectable option?

**Comment by ZachBacon** [comment]
Because not everyone wants a portable setup.

**Comment by rkitover** [comment]
Actually I think we should have a menu option to copy the config to the `.exe` location etc..

**Comment by rkitover** [comment]
Anyway I wanted to discuss this, that's why I told him to open the issue.

**Comment by denisfa** [comment]
Being portable does not make much sense on Linux, but it is very natural on windows. Maybe on macos too? We already have something like "search first on current dir" for our configuration. We could have an option on windows  to use portable mode (ticked by default) on `Options` > `General` or on `Help` menu.

**Comment by rkitover** [comment]
Yeah, the issue is that going into the `AppData` directories on Windows is something only advanced Windows users are familiar with. Which is why I added the `Factory Reset` menu item under `Help`. All that does is delete the config in `AppData`.

For mac we need to tweak our config finding a bit to support this better, we need to look not in the executable location, but the `.app` bundle location, which is a couple levels below.

But this is mainly something Windows users want.

So here's my idea, under the `General` config dialog, we can add a `Config Location` pane:

```
Config Location

AppData                     [check box]
.exe dir                    [check box]
```

The state of the check box would be "config exists in this location". Pressing the check box would copy the config there, clearing it would delete it from there.

So to make your installation "portable", all you'd do is open that pref pane and press `copy` for the `.exe` directory.

If your installation is "portable" but you want to copy the config to `AppData` you could do that as well.

And from here you could delete the config from either or both locations.

**Comment by Liger0** [comment]
Using as default the exe location would be better also because, in case you already have general settings on your pc for the other versions but you want one that needs not to use them, you can reset its own config file safely without having to remove the appdata one.
If you make it portable by transferring the config file into the portable one you would already have some predefined settings, and deleting the config file would make it in the appdata folder and the "loop" would continue. 
Maybe a way to circumvent this would be to create an empty .ini in the portable one and erase the content manually each time to avoid cancelling the appdata one that would interfere with other buils on the PC. 
Not sure if vba will rewrite the config with all the missing options tho and this is very tricky.
This would be valid on all the OS, not just windows.

Using as default the portable option and storing the config in the OS only if selected in the options would be the safer approach for any use case, imo.

**Comment by rkitover** [comment]
@Liger0 sorry, I don't understand what you mean.

If there is a config in the .exe folder then the one in appdata has no effect, the first one found is used.

**Comment by Liger0** [comment]
Yeah, but you may want to restart it from scratch rather than using one. Deleting one from the exe folder won't make the settings clean but it will just make it use again the  appdata one.

**Comment by rkitover** [comment]
So I already have "factory reset" to remove the appdata config, so what do you think about my "config location" pref pane I described above, there if you remove both checkboxes then both configs would be removed.

**Comment by rkitover** [comment]
So I would also change "factory reset" to remove ALL configs, not just the first one found.

**Comment by Liger0** [comment]
Maybe automatically creating a backup .bak file of the appdata would be a good idea to avoid any removing by mistake.

**Comment by ZachBacon** [comment]
@rkitover maybe change it to a back up and restore along with a full on factory reset. The back up and restore for quick testing etc

**Comment by rkitover** [comment]
Yeah we can also add the backup/restore to the new pref pane.

@denisfa what do you think about all of this.

**Comment by phs** [comment]
As a slightly different take o the issue, I just compiled vbam 2.1.4 SDL from source on linux and was hoping to find (say) a command line option to generate a new blank/default/factory config file.

If a new UI element appears to generate such a config, then CLI support would be appreciated as well.
